From c33fdb40ac2645515d6504c023dc4f0c114636fc Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 08 Jan 2019 15:44:07 +0200
Subject: [PATCH] [SW-0] pll: configure CPU PLL only if in bypass

Currently we are not configuring the CPU PLL at all as a w/a to
stability issue.
While working on palladium, or just loading bootloader components
directly to memory (with JTAG), the CPU PLL does not get configured.
Add CPU PLL configuration depending if it is in bypass
or already configured.

Change-Id: Ic1735062cd99c9c54094371ed4c2bad7a4cf40a9
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/pll.c b/pll.c
index 0c3e39a..1c2fa6c 100644
--- a/pll.c
+++ b/pll.c
@@ -231,7 +231,12 @@
 
 	pll_num = hl_pll_get_cfg(&hl_pll_cfg);
 
-	for (i=1; i<pll_num; i++)
+	for (i = 0; i < pll_num; i++) {
+		if ((u64)hl_pll_cfg[i].base == mmCPU_PLL_BASE &&
+			!hl_pll_is_bypass(hl_pll_cfg + i, HL_PLL_DIV0))
+				continue;
+
 		hl_pll_set(hl_pll_cfg + i);
+	}
 }
 #endif
